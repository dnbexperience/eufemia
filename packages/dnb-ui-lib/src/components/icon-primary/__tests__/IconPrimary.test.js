/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent
} from '../../../core/jest/jestSetup'
import Component from '../IconPrimary'

const props = fakeProps(require.resolve('../../icon/IconPrimary'), {
  optional: true
})
props.icon = 'question'

describe('IconPrimary component', () => {
  it('have to match snapshot', () => {
    const Comp = mount(<Component {...props} />)
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('has valid width and height prop', () => {
    const width = 200
    const height = 100
    const Comp = mount(
      <Component {...props} width={width} height={height} />
    )
    const elem = Comp.find('svg')
    expect(elem.exists()).toBe(true)
    expect(elem.props().width).toBe(width)
    expect(elem.props().height).toBe(height)
  })

  it('has valid medium size', () => {
    // here we explicit set size="medium" as well, cause we then test that the loadSVG makes a good job
    const Comp = mount(
      <Component {...props} icon="question_medium" size="medium" />
    )
    const svg = Comp.find('svg')
    const path = svg.find('path')
    expect(svg.exists()).toBe(true)
    expect(path.exists()).toBe(true)
    expect(svg.props().viewBox).toBe('0 0 24 24')
  })

  it('should validate with ARIA rules', async () => {
    const Comp = mount(<Component {...props} />)
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})
