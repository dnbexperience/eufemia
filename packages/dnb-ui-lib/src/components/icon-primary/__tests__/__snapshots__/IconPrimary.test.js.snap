// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconPrimary component have to match snapshot 1`] = `
<IconPrimary
  alt={null}
  attributes={null}
  border={null}
  class={null}
  className={null}
  color={null}
  height={null}
  icon="question"
  modifier={null}
  size={null}
  title={null}
  width={null}
>
  <span
    alt="question"
    aria-label="question"
    className="dnb-icon"
    role="img"
  >
    <span>
      <question>
        <svg
          fill="none"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            clipRule="evenodd"
            d="M4.352 5.1c0-2.05 1.55-3.6 3.6-3.6 2.052 0 3.601 1.55 3.601 3.6 0 2.051-1.55 3.6-3.6 3.6a.75.75 0 00-.75.75v2.666a.75.75 0 001.5 0V10.15c2.506-.348 4.35-2.425 4.35-5.049 0-2.879-2.22-5.1-5.1-5.1-2.88 0-5.1 2.221-5.1 5.1a.75.75 0 001.5 0zm3.6 10.898c.659 0 1.098-.439 1.098-1.097 0-.658-.44-1.097-1.097-1.097-.658 0-1.097.44-1.097 1.097 0 .658.439 1.097 1.097 1.097z"
            fill="#000"
            fillRule="evenodd"
          />
        </svg>
      </question>
    </span>
  </span>
</IconPrimary>
`;
